src/scripts/main.js

```diff
@@ -1,3 +1,279 @@
 'use strict';
 
 // write code here
+
+const pageTable = document.querySelector('table');
+
+const pageTableState = {};
+
+function extractAmount(str) {
+  return parseFloat(str.replace(/[^0-9.]/g, ''));
+}
+
+function convertToCurrency(val) {
+  return val.toLocaleString('en-US', {
+    style: 'currency',
+    currency: 'USD',
+    minimumFractionDigits: 0,
+    maximumFractionDigits: 0,
+  });
+}
+
+function validateEmployee(employee) {
+  const { employeeName, position, office, age, salary } = employee;
+
+  if (
+    !employeeName ||
+    employeeName.length < 4 ||
+    !position ||
+    !office ||
+    +age < 18 ||
+    +age > 90 ||
+    !salary ||
+    Number.isNaN(+salary)
```

Comment:
It would be nice to show a bit more informative errors in each case depending on which value is wrong, for example: `Age is required` or `Allowed age is between 18 and 90`, etc